### PR TITLE
Refactor blit copies and reuse scalar tensor's MPS storage in BinaryOps 

### PR DESCRIPTION
**Pull request by razarmehr**

### CODE REVIEW

File: aten/src/ATen/mps/MPSStream.mm [hunk 1]
```diff
@@ -46,17 +48,32 @@
...
     flush();
   }
+#endif
 }
 
 void MPSStream::commitAndWait() {
   assert(_commandBuffer);
+#if USE_MPSCOMMANDBUFFER
+  id<MTLCommandBuffer> currentCommandBuffer = _commandBuffer.rootCommandBuffer;
+  [_commandBuffer commitAndContinue];
```
[the remark this answers is not in the document]
Comment (kulinseth) [reply aten/src/ATen/mps/MPSStream.mm:64]: We can't just `commit` this, we need to wait as its blocking copy.

[the remark this answers is not in the document]
Comment (kulinseth) [reply aten/src/ATen/mps/MPSStream.mm:64]: Oh are you saying the code as was before is better?, that I agree.

[the remark this answers is not in the document]
Comment (saharshoza) [reply aten/src/ATen/mps/MPSStream.mm:64]: Yes. I think this function should be left as is.

**Comment by razarmehr** [comment]
Code is only refactoring. Will close it.

File: aten/src/ATen/mps/MPSStream.mm [hunk 2]
```diff
@@ -46,9 +48,13 @@
 }
 
 void MPSStream::commit(bool doFlush) {
+#if USE_MPSCOMMANDBUFFER
+  [commandBuffer() commitAndContinue];
+#else
   if (doFlush) {
```
Comment (kulinseth) [inline aten/src/ATen/mps/MPSStream.mm:54]: If this is always true, we should remove this condition.

**Review by kulinseth — APPROVED** [review]
Otherwise looks fine.

File: aten/src/ATen/mps/MPSStream.mm (hunk 2 above)
Comment (razarmehr) [reply aten/src/ATen/mps/MPSStream.mm:54]: Yes, the doFlush is always true.

**Pull request merged**